internal/controllers/extension_controller.go

```diff
@@ -512,7 +512,14 @@ func (r *ExtensionReconciler) resolve(ctx context.Context, extension ocv1alpha1.
 		}
 		if installedVersionSemver != nil {
 			installedVersion = installedVersionSemver.String()
-			predicates = append(predicates, catalogfilter.HigherBundleVersion(installedVersionSemver))
```

Comment:
With the current code there are a few issues:

1. `catalogfilter.HigherBundleVersion` is not really required as we already have more generic `catalogfilter.InMastermindsSemverRange`.
1. I think `catalogfilter.HigherBundleVersion` allows upgrades from one major version to another major version, etc. This is not the same behaviour as in `ClusterExtension`
1. We do not seem to support legacy `replaces` here.

This PR solves first two. Third still need to be addressed.

How it currently works in `ClusterExtension`: https://github.com/operator-framework/operator-controller/blob/52cb7eb1a650731f2e16253a7f0f0ba25ea3af70/docs/drafts/upgrade-support.md